nimble/host/include/host/ble_cs.h

```diff
@@ -0,0 +1,352 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef H_BLE_CS_
+#define H_BLE_CS_
+#include "syscfg/syscfg.h"
+
+struct ble_cs_rd_loc_supp_cap_rp {
```

Comment:
seems like duplicate of hci_common.h - keep it there

Reply:
So is it OK to use a structure from hci_common.h in the CS public api? If this is fine, I would get rid of the other duplicate structures in this file and the functions would just translate the endianness of some fileds, like this:
```C
int
ble_cs_rd_loc_supp_cap(struct ble_hci_le_cs_rd_loc_supp_cap_rp *rp)
{
    int rc;

    rc = ble_hs_hci_cmd_tx(BLE_HCI_OP(BLE_HCI_OGF_LE,
                                      BLE_HCI_OCF_LE_CS_RD_LOC_SUPP_CAP),
                           NULL, 0, rp, sizeof(*rp));

    rp->max_consecutive_procedures_supported = le16toh(rp->max_consecutive_procedures_supported);
    rp->optional_nadm_sounding_capability = le16toh(rp->optional_nadm_sounding_capability);
    rp->optional_nadm_random_capability = le16toh(rp->optional_nadm_random_capability);
    rp->optional_subfeatures_supported = le16toh(rp->optional_subfeatures_supported);
    rp->optional_t_ip1_times_supported = le16toh(rp->optional_t_ip1_times_supported);
    rp->optional_t_ip2_times_supported = le16toh(rp->optional_t_ip2_times_supported);
    rp->optional_t_fcs_times_supported = le16toh(rp->optional_t_fcs_times_supported);
    rp->optional_t_pm_times_supported = le16toh(rp->optional_t_pm_times_supported);

    return rc;
}
```

Reply:
functions like that should be just static in ble_cs.c
public API should not be wrappers for HCI commands, we should try to hide some parts that are irrelevant for user if possible. at the moment I wouldn't focus too much on that API, just make something that allow to interact with CS and we can think of it later. simply put a comment in ble_cs.h that all APIs in that file are experimental and subject to change at any time.